platform/ios/src/MGLMapView.mm

```diff
@@ -1619,6 +1748,73 @@ - (void)handleTwoFingerDragGesture:(UIPanGestureRecognizer *)twoFingerDrag
         [self notifyGestureDidEndWithDrift:NO];
         [self unrotateIfNeededForGesture];
     }
+
+}
+
+- (MGLMapCamera *)currentCameraWithEstimatedEndPoint:(CGPoint)endPoint panGesture:(UIPanGestureRecognizer *)pan
+{
+    MGLMapCamera *panCamera = [self.camera copy];
+    CGFloat width = CGRectGetWidth(self.bounds);
+    CGFloat height = CGRectGetHeight(self.bounds);
+    
+    mbgl::ScreenCoordinate screenCenter = {
+        width/2.,
+        height/2.,
+    };
+    
+    mbgl::ScreenCoordinate centerOffset = {
+        endPoint.x,
+        -endPoint.y,
+    };
+    
+    mbgl::ScreenCoordinate centerPoint = screenCenter - centerOffset;
```

Comment:
You can simplify this logic by creating a `CGPoint` directly instead of a `ScreenCoordinate`:

```objc
CGPoint centerPoint = CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds));
```